libr/debug/p/native/linux/linux_debug.c

```diff
@@ -509,3 +514,1397 @@ RList *linux_desc_list (int pid) {
 	closedir (dd);
 	return ret;
 }
+
+
+/* Coredump  functions */
+
+static map_file_t mapping_file = {0,0};
+
+static char *prpsinfo_get_fname(FILE *f) {
+        char *p;
+        int c;
+        int pos;
+
+        p = malloc(16);
+        pos = 0;
+        while ((c = fgetc(f)) != EOF && c != '\0')
```

Comment:
Dont use file operations there. This is bypassing the sandbox. We have util/file.c which all ops check for sandbox()

> On 13 May 2016, at 23:46, Riccardo Schirone <notifications@github.com> wrote:
> 
> In libr/debug/p/native/linux/linux_debug.c:
> 
> > @@ -509,3 +514,1397 @@ RList *linux_desc_list (int pid) {
> >  	closedir (dd);
> >  	return ret;
> >  }
> > +
> > +
> > +/* Coredump  functions */
> > +
> > +static map_file_t mapping_file = {0,0};
> > +
> > +static char *prpsinfo_get_fname(FILE *f) {
> > +        char *p;
> > +        int c;
> > +        int pos;
> > +
> > +        p = malloc(16);
> > +        pos = 0;
> > +        while ((c = fgetc(f)) != EOF && c != '\0')
> why not an fread? If that can't work, can we at least put a limit on the maximum number of chars that can be read with this loop?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
>